clickhouse/client.h

```diff
@@ -80,6 +80,11 @@ class Client {
     /// one or more call of \p cb.
     void Select(const std::string& query, SelectCallback cb);
 
+    /// Intends for execute select queries.  Data will be returned with
+    /// one or more call of \p cb.
+    /// extraInfo will be passed to callback
+    void Select(const std::string& query, void* extraInfo, SelectCallbackWithInfo cb);
+
```

Comment:
Since SelectCallbackWithInfo is a std::function<> why you cannot use variable capture with lambdas and write something like this?
```
client.Select("SELECT * FROM test.test", [extraInfo](const Block& block) {
  // use extraInfo here
});
```